File: web/types/statistics.ts
import { UserBloodPressure } from "./user-blood-pressure";
import { WeightRecord } from "./weight-record";
import { UserTemperature } from "./user-temperature";

export interface StatisticsSummary {
  latestBloodPressure: UserBloodPressure | null;
  latestWeight: WeightRecord | null;
  latestTemperature: UserTemperature | null;
  latestHeight: { height: number; recordDate: string } | null;
  bloodPressureTrend: UserBloodPressure[];
  weightTrend: WeightRecord[];
  temperatureTrend: UserTemperature[];
}

File: internal/model/statistics.go
package model

type StatisticsSummary struct {
	LatestBloodPressure *UserBloodPressure      `json:"latestBloodPressure"`
	LatestWeight        *WeightRecord           `json:"latestWeight"`
	LatestTemperature   *UserTemperatureRecord  `json:"latestTemperature"`
	LatestHeight        *UserHeightVO           `json:"latestHeight"`
	BloodPressureTrend  []*UserBloodPressure    `json:"bloodPressureTrend"`
	WeightTrend         []*WeightRecord         `json:"weightTrend"`
	TemperatureTrend    []*UserTemperature      `json:"temperatureTrend"`
}
